polaris-react/src/styles/_common.scss

```diff
@@ -23,4 +23,7 @@
 // tag+attribute selector.
 // However, we're still supporting iOS 12 for embedded web views in some apps, which
 // unfortunately doesn't support :where https://caniuse.com/?search=%3Awhere
-$se23: 'html[class*="Polaris-Summer-Editions-2023"]';
+$se23: 'html[class~="Polaris-Summer-Editions-2023"]';
+$se23ShadowBevelOptOut: 'html[class~="Polaris-Summer-Editions-2023-Shadow-Bevel-Opt-Out"]';
```

Comment:
Adjusted the beta-flag attribute selectors from a partial match (`*=`) to a space delimited exact match (`~=`): https://github.com/Shopify/polaris/pull/9713/commits/f398dc66894bc69274ddd088f2d92c970f10e0b6

This fixes a bug where the shadow-bevel opt out flag enabled the se23 styles:

Before:

https://github.com/Shopify/polaris/assets/32409546/cfb8d8d0-7b37-450d-951e-79417010b5ae

After: 

https://github.com/Shopify/polaris/assets/32409546/260e45df-fd0d-4445-a7a0-ba490089619f

cc @kyledurand